core/src/components/picker-column-internal/picker-column-internal.tsx

```diff
@@ -228,10 +228,53 @@ export class PickerColumnInternal implements ComponentInterface {
   private centerPickerItemInView = (target: HTMLElement, smooth = true) => {
     const { el, isColumnVisible } = this;
     if (isColumnVisible) {
-      // (Vertical offset from parent) - (three empty picker rows) + (half the height of the target to ensure the scroll triggers)
-      const top = target.offsetTop - 3 * target.clientHeight + target.clientHeight / 2;
+      const targetBox = target.getBoundingClientRect();
+      const scrollBox = el.getBoundingClientRect();
+
+      /**
+       * Compute the desired offset location.
+       * This is the point where the element
+       * should be in order for it to appear
+       * centered within the column viewport.
+       */
+      const midpoint = scrollBox.height / 2 + targetBox.height / 2;
+
+      /**
+       * Use data-index to determine which
+       * number button this is. This number will
+       * be used to determine how far this button
+       * is offset from the top of the column.
+       * We add 1 to convert from a zero indexed
+       * value.
+       */
+      if (!target.hasAttribute('data-index')) {
+        return;
+      }
+      const index = target.getAttribute('data-index') as string;
+      const numberButton = parseInt(index) + 1;
+      if (!Number.isInteger(numberButton)) {
+        return;
+      }
+
+      /**
+       * From here, we determine how far
+       * this button offset from the top.
+       * We add 3 to numberButton to account
+       * for the three .picker-item-empty elements.
+       */
+      const targetOffsetFromTop = (numberButton + 3) * targetBox.height;
+
+      /**
+       * Compute the scroll position necessary
+       * for the button to be centered within the view.
+       * Convert the value to an integer to avoid half
+       * pixel scrolls which can be inconsistent across
+       * browsers.
+       */
+      const top = Math.round(targetOffsetFromTop - midpoint);
```

Comment:
https://user-images.githubusercontent.com/90629384/192820135-e2c14721-88fb-4aed-a157-eea30811f826.mp4 I'm seeing a weird issue where the time picker on the basic datetime opens with the wheels in the wrong positions, and then things jump around when I make the first click.